pkg/backend/openshiftcluster/delete.go

```diff
@@ -54,22 +54,17 @@ func (m *Manager) Delete(ctx context.Context) error {
 		}
 	}
 
-	_, err = m.groups.CheckExistence(ctx, m.doc.OpenShiftCluster.Properties.ResourceGroup)
-	if err != nil {
-		if err, ok := err.(autorest.DetailedError); ok {
-			if err.StatusCode == http.StatusForbidden {
-				return nil
-			}
-		}
-		return err
-	}
-
 	m.log.Print("deleting private endpoint")
 	err = m.privateendpoint.Delete(ctx, m.doc)
 	if err != nil {
 		return err
 	}
 
 	m.log.Printf("deleting resource group %s", m.doc.OpenShiftCluster.Properties.ResourceGroup)
-	return m.groups.DeleteAndWait(ctx, m.doc.OpenShiftCluster.Properties.ResourceGroup)
+	err = m.groups.DeleteAndWait(ctx, m.doc.OpenShiftCluster.Properties.ResourceGroup)
+	if detailedErr, ok := err.(autorest.DetailedError); ok &&
+		detailedErr.StatusCode == http.StatusForbidden {
+		err = nil
```

Comment:
Could you please explain why we say that there is no error if we can't remove the resource group?

Reply:
Yes: backend functions should be restartable.  Say a delete is running, but that RP crashes and restarts: by the time it gets here, the group could be gone.  If so, that shouldn't be an error.

I'm not currently 100% sure if the error returned in dev == the error in prod, and I'm also not sure if to expect forbidden or notfound - will check as a followup

Reply:
Got it. Thanks.

Reply:
Yes, slightly non-intuitively, it's Forbidden.